addons/project/models/project.py

```diff
@@ -1176,10 +1288,72 @@ def default_get(self, default_fields):
 
         return vals
 
+    def read(self, fields=None, load='_classic_read'):
+        if fields and self.env.user.has_group('base.group_portal'):
+            for key in fields:
+                if key not in self.SELF_READABLE_FIELDS:
+                    raise AccessError(_('You cannot read %s field in task.') % key)
```

Comment:
Sould probably `', '.join()` the inaccessible fields as otherwise it'll be displayed as a set literal which would look rather odd to the reader / user.

Reply:
Note that in an ideal word, this should never be displayed to the honest user, only in case of somebody trying to cheat via rpc or in case of bugs.